README.md

```diff
@@ -1,125 +1,167 @@
-# mason-null-ls
+`mason-null-ls` bridges `mason.nvim` with the `null-ls` plugin - making it easier to use both plugins together.
 
-Install or upgrade the required tools needed for null-ls where available in Mason.
 
-Currently installation of the tools must occur using a command on startup. 
+# Introduction
 
-`require("mason-null-ls").check_install()`
+`mason-null-ls.nvim` closes some gaps that exist between `mason.nvim` and `null-ls`. Its main responsibilities are:
 
-Uses Mason do the heavy lifting of installing the dependencies.
+-   provide extra convenience APIs such as the `:NullLsInstall` command
+-   allow you to (i) automatically install, and (ii) automatically set up a predefined list of sources
+-   translate between `null-ls` source names and `mason.nvim` package names (e.g. `haml_lint` <-> `haml-lint`)
 
-**Can automatically parse the null-ls table and install the necessary tools available in Mason.**
+It is recommended to use this extension if you use `mason.nvim` and `null-ls`.
 
-Note: Going through all null-ls builtins and mapping it to the correct tool name in Mason is still a WIP.
-Feel free to create an issue or PR with a source you would like available immediately.
+**Note: this plugin uses the `null-ls` source names in the APIs it exposes - not `mason.nvim` package names.
```

Comment:
what are reasons for removing the feature?

Reply:
the S in **S**OLID?

Reply:
Sorry I don't understand what you're exactly referring to here, could you elaborate further.
In any case if this about the README feel free to edit it however you see fit.

Reply:
oh I meant you were able to install both `mason tools` by name and using `null-ls sources` before. Now you can only do the second. Was there a reason for that?

Reply:
Ahaa, figured some conformity would do us good here.
`MasonInstall` installs mason packages
`LspInstall` installs lsp servers
`NullInstall` installs null sources